hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -403,6 +426,52 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 			log.Info("hostedcluster is still deleting", "name", req.NamespacedName)
 			return ctrl.Result{RequeueAfter: clusterDeletionRequeueDuration}, nil
 		}
+
+		// This new condition is necessary for OCM personnel to report any cloud dangling objects to the user.
+		// The grace period is customizable using an annotation called HCDestroyGracePeriodAnnotation. It's a time.Duration annotation.
+		// This annotation will create a new condition called HostedClusterDestroyed which in conjuntion with CloudResourcesDestroyed
```

Comment:
>This annotation will create a new condition called HostedClusterDestroyed which in conjuntion with CloudResourcesDestroyed

It's very difficult to me to reason about that statement. What's the first condition giving over the second?
Can you share a matrix with both conditions possible values and what that means for a consumer.

I don't follow why this PR needs to be something else than just adjusting this lines of code for a configurable timeout instead of a hardcoded one This annotation will create a new condition called HostedClusterDestroyed which in conjuntion with CloudResourcesDestroyed

Reply:
> HostedClusterDestroyed.Status == true && CloudResourcesDestroyed.Status == true

This is the happy path, no? It's what normally should happen.

Reply:
yes

Reply:
How [is this better](https://github.com/openshift/hypershift/pull/2894/files#diff-b5cde720842b19b53da3f454d827e27327c4559b2e4849906dac728586f51eb3R429-R456) than just adjusting [the existing timeout](https://github.com/openshift/hypershift/blob/24cd25ce3fcbe05ced7a67aeaa4101a433dafa38/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L3742-L3749) to honour an annotation and let that be the grace period?

Reply:
That timeout is used to give up on resource deletion and I'm using both resource deletion + grace period that I think they should be decoupled.

Once the resourceDeletion timeout have been triggered and the HCP has been deleted, the grace period starts and a new condition is created. If we just go with the existing timeout, the OCM personnel will not have time to grab the condition.